docs/guides/c-extension-for-python.mdx

```diff
@@ -0,0 +1,230 @@
+
+PyObject *build_observable(void) {
+    QkObs *obs = qk_obs_zero(100);
+    // build the observable ...
+    PyObject *pyobj = qk_obs_to_python(obs); // convert to Qiskit's Python ``SparseObservable``
+    qk_obs_free(obs);
+    return pyobj;
+}
+```
+
+Below we will show how to compile this into a shared library, say `qiskit_cextension.so`.
+Once this is done, we will be able to call the C program from Python as
+```python
+# file: main.py
+import qiskit
+import ctypes
+
+# Load the extension, ensuring the global interpreter lock (GIL) is acquired for function calls,
+# which we need for the C->Python object conversion.
+lib = ctypes.PyDLL("/path/to/qiskit_cextension.so")
+lib.build_observable.argtypes = None  # set argument types to the function
+lib.build_observable.restype = ctypes.py_object  # set return type
+
+# now we can directly call the function
+obs = lib.build_observable()
+print("SparseObservable instance?", isinstance(obs, qiskit.quantum_info.SparseObservable))
+print(obs)
+```
+
+## Build
+
+First, you have to build the Qiskit Python extension. This includes the C symbols so you can access both interfaces via the same shared library. This is important to ensure data can be passed
+correctly across C and Python.
+```bash
+python setup.py build_rust --inplace --release
+```
+The shared library is called ``_accelerate.<platform-specific-part>``, whose location and name we can find as
+```bash
+QKLIB=$(python -c "import os; import qiskit; print(os.path.dirname(qiskit._accelerate.__file__))")
+QKNAME=$(python -c "import os; import qiskit; print(os.path.basename(qiskit._accelerate.__file__))")
+```
+
+We will need to know the location of the environments Python includes (``Python.h``) and libraries (``libpython.<suffix>``).
+These can, for example, be identified via
+```bash
+PYINCLUDE=$(python -c "import sysconfig; print(sysconfig.get_path('include'))")
+PYLIB=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
+PYNAME=$(find $PYLIB -maxdepth 1 -name "libpython*" | grep -oE "[^/]+$" | grep -oE "python[0-9]+\.[0-9]+" || echo "python")
+```
+(If you just know these locations and names, you can also just set them directly.)
+
+### Linux
+
+Linking can slightly differ between platforms, here we first describe a solution for linkers
+supporting libraries with arbitrary names, using the `-l:` flag. See below if your linker
+requires the library to be called `lib<something>`.
+
+We can build the extension specifying the full name of the `_accelerate` library:
+```bash
+gcc extension.c -fpic -shared -o cextension.so \
+  -I/path/to/dist/c/include -L$QKLIB -l:$QKNAME \
+  -I$PYINCLUDE -L$PYLIB -l$PYNAME
+```
+And then simply ``python main.py`` to run the Python program.
+
+### MacOS
+
+An alternative to using the exact library name with `-l:` is to symlink the `_accelerate` library
+to the desired name. This solution works on Mac (and also Linux if you replace all occurences of `.dylib` with `.so`).
```

Comment:
Should the Linux workaround be mentioned in the Linux section above?

Reply:
We shouldn't need it on linux, the above solution should always work there 🙂 It would just technically be possible to also use this symlink approach, but it's a suboptimal solution that we wouldn't want to encourage, I think 😄

Reply:
got it, thanks!